content/features/security-privacy/security/membership-providers/index.md

```diff
@@ -0,0 +1,128 @@
+- [Understanding ASP.NET Membership](https://docs.microsoft.com/en-us/previous-versions/aspnet/tw292whz(v=vs.100))
+- [ASP.NET Membership Provider Toolkit](https://docs.microsoft.com/en-us/previous-versions/aspnet/6e9y4s5t(v=vs.100))
+
+### Custom Providers
+Developers can create custom membership providers by implementing the `MembershipProvider` abstract class and configuring them in the DNN provider configuration.
+
+## Configuration
+
+Membership providers are configured in the `web.config` file under the `system.web/membership` section. The default AspNetSqlMembershipProvider is configured automatically during DNN installation.
+
+## Force Password Reset
+
+There are scenarios where administrators need to force all users to reset their passwords, such as:
+
+- After upgrading password storage formats (e.g., from Encrypted to Hashed)
+- Following a security incident or suspected breach
+- Implementing new password policies
+
+### Migration from "encrypted" to "hashed"
+To change the password format in DNN from encrypted to hashed, you must edit the web.config file and change the passwordFormat attribute to `Hashed` within the <add name="AspNetSqlMembershipProvider" ... /> section, and simultaneously set `enablePasswordRetrieval` to `false`. After this change, existing users' encrypted passwords will be converted to hashed passwords when they update their password.
+
+To force all users to reset their passwords, you can execute the following SQL query against your DNN database:
+
+> ⚠️ **Important**: Always backup your database before executing any direct SQL modifications. Also, the default method to reset a password involves sending a verification link by email. Make sure emails work correctly and that your own superuser acccount has a proper email where you can receive that link.
+
+```sql
+-- Force password reset for all users
+UPDATE Users 
+SET UpdatePassword = 1
+```
+
+### Changing only the hashing algorithm
+
+> ⚠️ **Warning**: Because hashed passwords cannot be decrypted, this change will prevent any logins with the existing passwords (including super-users), which may be confusing for users. To help avoid confusion you may want to notify all your users about having to reset their passwords for better security. They will have to click on "Reset Password" to migrate to the new format. They will be able to enter their username and receive a special link by email to reset their password using a token.
+
+> 💡You can check the `LastPasswordChangedDate` in the `aspnet_Membership` table to see which users did change their passwords or not after the date of that change. You may use that information to later delete users that may no longer be activivally engaged. Additionally you can wipe the `Password` field if you want to make sure no passwords with the old algorithm are kept (before notifying users about the change).
+
+> 💡**Is it critical to migrate from Encrypted to Hashed?**
```

Comment:
I think this should be moved above the heading on line 117 to be with the section about changing from Encrypted -> Hashed.